jcl/src/java.base/share/classes/openj9/internal/tools/attach/target/DiagnosticUtils.java

```diff
@@ -273,12 +273,37 @@ private static DiagnosticProperties doDump(String diagnosticCommand) {
 			}
 			if (!dumpType.isEmpty()) {
 				String request = dumpType;
-				if (parts.length == 2) {
-					String fileDirective = ("system".equals(dumpType) && IPC.isZOS) ? ":dsn=" : ":file="; //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$
-					request += fileDirective + parts[1];
+				boolean foundSystemRequest = false;
+				boolean heapDump = "heap".equals(dumpType);
+				boolean systemDump = "system".equals(dumpType);
+				String options = ""; //$NON-NLS-1$
+				String separator = ":"; //$NON-NLS-1$
+				for (int i = 1; i < parts.length; i++) {
+					String option = parts[i];
+					boolean isRequest = option.startsWith("request=");
+					boolean isOpts = option.startsWith("opts=");
+					if (isRequest || isOpts) { //$NON-NLS-1$ //$NON-NLS-2$
+						if (heapDump && isOpts) {
```

Comment:
Shouldn't this be `systemDump && isOpts` instead?

Reply:
Fixed it as !dumpDump && isOpts